src/game/World.cpp

```diff
@@ -2561,10 +2561,13 @@ BanReturn World::BanAccount(BanMode mode, std::string nameOrIP, uint32 duration_
             break;
         }
         default:
+            delete holder;
+
             return BAN_SYNTAX_ERROR;
     }
 
     db->DelayQueryHolderUnsafe(&banHandler, &BanAccountHandler::HandleAccountSelectResult, holder);
+    delete holder;
```

Comment:
The function called right before this says "Delay" in the name. Are you sure it's not storing an using the holder later on?